src/router/index.ts

```diff
@@ -42,14 +48,21 @@ const router = createRouter({
 });
 
 router.beforeEach((to) => {
-  if (
-    to.meta.requiresAuth &&
-    (to.query.userId == undefined || to.query.userId == "")
-  ) {
-    return {
-      name: "403",
-    };
+  /** Before each router, check if the user is a third party and therefore, needs authentication. */
+  if (to.meta.requiresAuth) {
+    const queryParams = Object.keys(to.query);
+    const isAuthNeeded =
+      requiredAuthKeys.every((key) => queryParams.includes(key)) &&
+      queryParams.every(
+        (key) => to.query[key] != "" && to.query[key] != undefined
+      );
```

Comment:
What we are really checking is whether the user is authenticated right?

```suggestion
    const isAuthenticated =
      requiredAuthKeys.every((key) => queryParams.includes(key) &&  (queryParams[key] != "" && queryParams[key] != undefined))
```